src/test/kotlin/com/eh/niver/service/impl/GroupServiceImplTest.kt

```diff
@@ -0,0 +1,296 @@
+            idPerson = 8,
+            name = "Joselito",
+            birthday = LocalDate.now(),
+            email = "teste@teste.com",
+            password = "joselito123"
+        )
+
+        val groupReturn = Group(
+            idGroup = null,
+            name = "Teste",
+            owner = owner,
+            members = mutableListOf(owner)
+        )
+
+        every { personService.getPersonById(any()) } returns owner
+        every { repository.save(any()) } returns groupReturn
+
+        val result = service.saveGroup(groupRequest)
+
+        Assertions.assertEquals(groupReturn,result)
+    }
+
+    @DisplayName("should return a exception if the group was not created")
+    @Test
+    fun saveGroup_ShouldReturnExceptionWhenFailed() {
+        val groupRequest = RequestSaveGroup(
+            idGroup = null,
+            name = "Teste",
+            owner = ResponseMember(
+                idPerson = 8,
+                name = "Joselito",
+                birthday = LocalDate.now()
+            )
+        )
+
+        val owner = Person(
+            idPerson = 8,
+            name = "Joselito",
+            birthday = LocalDate.now(),
+            email = "teste@teste.com",
+            password = "joselito123"
+        )
+
+        val groupReturn = Group(
+            idGroup = null,
+            name = "Teste",
+            owner = owner,
+            members = mutableListOf(owner)
+        )
+
+        every { personService.getPersonById(any()) } throws Exception()
+        every { repository.save(any()) } returns groupReturn
+
+        assertThrows<Exception> { service.saveGroup(groupRequest) }
+        verify(exactly = 0) { repository.save(any()) }
+    }
+
+    @DisplayName("should return a group by id")
+    @Test
+    fun getGroupById_ShouldReturnGroupById() {
+        val groupId = "8"
+
+        val owner = Person(
+            idPerson = 8,
+            name = "Joselito",
+            birthday = LocalDate.now(),
+            email = "teste@teste.com",
+            password = "joselito123"
+        )
+
+        val group = Optional.of(Group(
+            idGroup = 8,
+            name = "Teste",
+            owner = owner,
+            members = mutableListOf(owner),
+            invite = null
+        ))
+
+        val groupResult = ResponseGroup(
+            idGroup = 8,
+            name = "Teste",
+            owner = ResponseMember(
+                idPerson = 8,
+                name = "Joselito",
+                birthday = LocalDate.now()
+            ),
+            members = mutableListOf(owner).map { ResponseMember(
+                idPerson = 8,
+                name = "Joselito",
+                birthday = LocalDate.now()
+            ) }
+        )
+        every { repository.findById(any()) } returns group
+
+        val result = service.getGroupById(groupId)
+
+        Assertions.assertEquals(groupResult,result)
+    }
+
+    @DisplayName("should return exception when group is empty")
```

Comment:
sugestao para alterar o nome do teste para:
should return exception when group is not exists